src/som/interpreter/actors/EventualSendNode.java

```diff
@@ -175,21 +193,30 @@ protected void sendDirectMessage(final Object[] args, final Actor owner,
       assert !(args[0] instanceof SFarReference) : "This should not happen for this specialization, but it is handled in determineTargetAndWrapArguments(.)";
       assert !(args[0] instanceof SPromise) : "Should not happen either, but just to be sure";
 
+      SteppingType step = getStepping();
+
       DirectMessage msg = new DirectMessage(
-          EventualMessage.getCurrentExecutingMessageId(), target, selector, args,
+          EventualMessage.getCurrentExecutingMessage(), EventualMessage.getCurrentExecutingMessageId(), target, selector, args,
           owner, resolver, onReceive,
           messageReceiverBreakpoint.executeCheckIsSetAndEnabled(), promiseResolverBreakpoint.executeCheckIsSetAndEnabled());
+
+      msg.setTriggerStepping(step);
```

Comment:
Why do you set the `step` on the message after its creation, and don't pass it in the constructor?
Also, I guess the field in the message object could be final then?